src/verifier/verifier.rs

```diff
@@ -45,7 +45,7 @@ pub fn check_duplicate_id(
                     // happens if new_loc is compiler-provided or is a user-def func without location -- both should throw unexpected error
                     err.unexpected_error(true, Some(UNEXPECTED_ERR_MSG.to_string()), None);
                 } else {
-                    err.compiler_fn_overload_error(false, name.clone(), new_loc);
+                    err.compiler_fn_overload_error(false, name.to_owned(), new_loc);
```

Comment:
TLDR; I think this is okay for now, but we may have problems later.

I'm surprised this works...I'd think the `clone`s are necessary...otherwise you'd be moving the `name` var out of the original owner. Maybe it's okay since the AST is changed to a different representation and not used after the verifier stage? We may have problems with this once we work on targeting Wizard (I think we're going to have to emit by traversing the original AST).